CHANGELOG.md

```diff
@@ -34,6 +34,9 @@ This project adheres to [Semantic Versioning](http://semver.org/).
  - #3697, #3602, Handle queries on non-existing table gracefully - @taimoorzaeem
  - #3600, #3926, Improve JWT errors - @taimoorzaeem
  - #3013, Fix `order=` with POST, PATCH, PUT and DELETE requests - @taimoorzaeem
+ - #3498, Improve media type parser - @taimoorzaeem
+   + Allows space separated parameters
+   + Handles quoted text gracefully
```

Comment:
How about some tests for these fixes?

Reply:
So, we don't currently have this issue with `Accept` header because we use `parseHttpAccept` library function. But this issue is only for `Content-Type` header. I can't think of a good test `spec` with `Content-Type` as we currently don't use parameters for them (we ignore them). I have added the `doctests` for these fixes.

Later with #3164 this behavior will be tested with `Content-Type` header because then, we will use the parameters. So I think I should specify in the CHANGELOG that media parsing is improved for `Content-Type` header?

Reply:
> I still don't get is how is this a user-facing fix

Thinking more about it, I guess this isn't user facing at all **yet**. But it's benefit will be seen in #3164. I'll change this to a `refactor: ...` instead.

> Perhaps you can share some fixed cases here to help me understand?

Sure. So, previously we had this which breaks when their is a semicolon in quotes like `\" ; \"`:
```haskell
>>> tokenizeMediaType "application/vnd.pgrst.plan;wat=\"application/json;text/csv\""
("application","vnd.pgrst.plan",[("wat","application/json"),("text/csv\"","")])
```

This is fixed now with:

```haskell
>>> P.parse tokenizeMediaType "" "application/vnd.pgrst.plan;wat=\"application/json;text/csv\""
Right ("application","vnd.pgrst.plan",[("wat","application/json;text/csv")])
```

Reply:
Oh, I get it now. Given that the plan media type has the ability to include another media type inside:

> Accept: application/vnd.pgrst.plan; for="text/xml"
> From https://docs.postgrest.org/en/v12/references/observability.html#execution-plan

I think then this could be labeled as a fix for the plan media type. A `fix:` prefix is correct.

On the CHANGELOG it could be like:

```
Fix incorrect parsing of the `for` parameter on the ` application/vnd.pgrst.plan` media type.
```